azkaban-exec-server/src/main/java/azkaban/container/FlowContainer.java

```diff
@@ -0,0 +1,336 @@
+    this.projectDir = projectDirPath.toFile();
+
+    // Create Azkaban Props Map
+    this.azKabanProps = props;
+    // Setup global props if applicable
+    final String globalPropsPath = this.azKabanProps.getString("executor.global.properties", null);
+    if (globalPropsPath != null) {
+      this.globalProps = new Props(null, globalPropsPath);
+    }
+
+    // Setup DAO, a lot of it is redundant
+    final DataSource dataSource= new MySQLDataSource(this.azKabanProps,
+        new DBMetrics(new MetricsManager(new MetricRegistry())));
+
+    final DatabaseOperator dbOperator = new DatabaseOperator(
+        new QueryRunner(dataSource));
+
+    this.executorLoader = setupDao(dbOperator);
+
+    // project Loader
+    this.projectLoader = new JdbcProjectImpl(this.azKabanProps, dbOperator);
+
+    // setup executor service, TODO : revisit
+    this.executorService = Executors.newSingleThreadExecutor();
+
+    this.jobLogChunkSize = this.azKabanProps.getString("job.log.chunk.size", "5MB");
+    this.jobLogNumFiles = this.azKabanProps.getInt("job.log.backup.index", 4);
+    this.validateProxyUser = this.azKabanProps.getBoolean("proxy.user.lock.down", false);
+    this.jobTypeManager =
+        new JobTypeManager(
+            this.azKabanProps.getString(AzkabanExecutorServer.JOBTYPE_PLUGIN_DIR,
+                PluginManager.JOBTYPE_DEFAULTDIR),
+            this.globalProps, getClass().getClassLoader());
+    // Fetch keyStore props and use it get the KeyStore, put it in JobTypeManager
+    Props keyStoreLoadProps = this.jobTypeManager.getKeyStoreLoadProps();
+    if (keyStoreLoadProps != null) {
+      // Load HadoopSecurityManager
+      HadoopSecurityManager hadoopSecurityManager = null;
+      try {
+        final String hadoopSecurityClassName =
+                keyStoreLoadProps.getString(HadoopJobUtils.HADOOP_SECURITY_MANAGER_CLASS_PARAM);
+        final Class<?> hadoopSecurityManagerClass =
+                HadoopProxy.class.getClassLoader().loadClass(hadoopSecurityClassName);
+
+        logger.info("Loading hadoop security manager " + hadoopSecurityManagerClass.getName());
+        hadoopSecurityManager = (HadoopSecurityManager)
+                Utils.callConstructor(hadoopSecurityManagerClass, keyStoreLoadProps);
+      } catch (final Exception e) {
+        logger.error("Could not instantiate Hadoop Security Manager ", e);
+        throw new RuntimeException("Failed to get hadoop security manager!"
+                + e.getCause(), e);
+      }
+
+      final KeyStore keyStore = hadoopSecurityManager.getKeyStore(keyStoreLoadProps);
+      // Delete the cert file from disk
+      final File certFile = new File(Constants.ConfigurationKeys.CSR_KEYSTORE_LOCATION);
+      if (certFile.delete()) {
+        logger.info("Successfully deleted the cert file");
+      } else {
+        logger.error("Failed to delete the cert file");
+        throw new IOException("Failed to delete the cert file");
+      }
+      if (keyStore == null) {
```

Comment:
This should be right under the line number 167.

Reply:
Aha! If the keyStore is null, we abandon anyway.